includes/class-wc-stripe-webhook-handler.php

```diff
@@ -503,10 +503,10 @@ public function process_webhook_refund( $notification ) {
 	 * @param object $notification
 	 */
 	public function process_review_opened( $notification ) {
-		$order = WC_Stripe_Helper::get_order_by_charge_id( $notification->data->object->charge );
+		$order = WC_Stripe_Helper::get_order_by_intent_id( $notification->data->object->payment_intent );
 
 		if ( ! $order ) {
-			WC_Stripe_Logger::log( 'Could not find order via charge ID: ' . $notification->data->object->charge );
+			WC_Stripe_Logger::log( 'Could not find order via intent ID: ' . $notification->data->object->payment_intent );
```

Comment:
Is it at all possible for `$notification` to not have a `data`, for data to not have an `object`, or object to not have a `payment_intent`? (i.e. do we need to defend against a PHP notice here and below)?

Reply:
I spot-checked some of the other webhook handlers and they all assume `$notification->data->object` contains the expected object type. The only cases where that wouldn't be true would be:
- A forged webhook (in which case a PHP notice would be welcome IMHO)
- Closing a review that was opened before 4.2.0 (that is, before the Payment Intents were introduced). In that case, the webhook would contain a `charge` object instead of a `payment_intent`. But I don't think a review can just stay opened for months, so it's not worth coding that path.

Reply:
Not sure how big a deal it is, but thought it would be worth pointing out that anyone recently upgrading from <4.2.0 and then closing a review would run into the latter path.

Reply:
@dechov Good point! It somehow didn't occur to me that people can just skip over versions when they update. I've opened #919 with a quick fix.